### DISCUSSION TITLE
IgnoreHitPause in ZSS

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1342] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

From version 0.98.2 to 0.99, there was a change to make IgnoreHitPause work for an entire code block, instead of having to declare it repeatedly. I think this was a great change, but it's still not quite complete. I think IgnoreHitPause (and Persistent) should work for "else" blocks as well.

For instance, training.zss starts like this:
```
ignoreHitPause if gameMode != "training" || isHelper || teamSide != 2 {
	# Do nothing, not training more or statedef executed by helper or not P2
} else ignoreHitPause if roundState = 0 {
	# Round start reset
	powerSet{value: player(1),powerMax; redirectid: player(1),id}
	powerSet{value: player(2),powerMax; redirectid: player(2),id}
	map(_iksys_trainingLifeTimer) := 0;
	map(_iksys_trainingPowerTimer) := 0;
	map(_iksys_trainingDirection) := 0;
	map(_iksys_trainingAirJumpNum) := 0;
} else ignoreHitPause if roundState = 2 {
```

This still feels a bit redundant. The first IgnoreHitPause should probably be enough.
Then again we do already have the option to put the entire code inside an IgnoreHitPause declaration, which seems to work as intended.

I am not a professional, though, so I'm not sure what are the precedents here in coding languages. Thoughts?

**Comment by Lazin3ss** [comment, marked as answer]
I agree with this change, I'll add it soon

**Comment by K4thos** [comment]
it was already implemented by Lazin3ss
